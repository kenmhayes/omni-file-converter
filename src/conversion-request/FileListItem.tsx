import React, { useState } from 'react';
import { Anchor, Button } from 'react-bootstrap';
import deleteicon from '../assets/svg/x-circle.svg';
import { getS3ObjectUrl } from '../aws/S3Helper';

export interface FileListItemProps {
  fileName: string;
  onDelete: () => void;
}

function FileListItem(props: FileListItemProps) {
  const { fileName, onDelete } = props;
  const [downloadUrl, setDownloadUrl] = useState<string>('');

  const onButtonClick = async () => {
    // Temp code, so left untested
    const objectDownloadUrl = await getS3ObjectUrl(fileName);
    setDownloadUrl(objectDownloadUrl);
  };

  return (
    <div className="container">
      <div className="row">
        <div className="col-6">
          {fileName}
        </div>
        <div className="col-3">
          <Button onClick={onButtonClick}>Get Download</Button>
          <Anchor href={downloadUrl} disabled={!downloadUrl}>Download</Anchor>
        </div>
        <div className="col">
          <button type="button" aria-label="remove-file" className="btn" title="Remove File" onClick={onDelete}>
            <img src={deleteicon} alt="Remove File" width="32" height="32" />
          </button>
        </div>
      </div>
    </div>
  );
}

export default FileListItem;
