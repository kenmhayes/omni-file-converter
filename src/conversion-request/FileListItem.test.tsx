import React from 'react';
import { create } from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import FileListItem from './FileListItem';

it('renders a file list item', () => {
  const component = create(
    <FileListItem fileName="File" onDelete={jest.fn()}/>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls the delete handler when the delete icon is pressed', () => {
    const deleteHandler = jest.fn();

    const {queryByRole, getByRole } = render(
        <FileListItem fileName="File" onDelete={deleteHandler}/>
    );

    const deleteButtonRole = "button";
    const deleteButtonOptions = { name: /remove-file/i };
    expect(queryByRole(deleteButtonRole, deleteButtonOptions)).toBeTruthy();

    fireEvent.click(getByRole(deleteButtonRole, deleteButtonOptions));

    expect(deleteHandler).toHaveBeenCalled();
});

