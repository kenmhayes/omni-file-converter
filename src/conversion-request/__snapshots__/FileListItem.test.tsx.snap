// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a file list item 1`] = `
<div
  className="container"
>
  <div
    className="row align-items-center"
  >
    <div
      className="col-6"
    >
      File
    </div>
    <div
      className="col-3"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Get Download
      </button>
    </div>
    <div
      className="col"
    >
      <button
        aria-label="remove-file"
        className="btn"
        onClick={[MockFunction]}
        title="Remove File"
        type="button"
      >
        <img
          alt="Remove File"
          height="32"
          src="x-circle.svg"
          width="32"
        />
      </button>
    </div>
  </div>
</div>
`;
